design/proof-network.md

```diff
@@ -0,0 +1,257 @@
+transactions is done after sequencing. This means that storage proofs can be
+checked in parallel, which allows validators to scale up and support a large
+storage network that produces many proofs.
+
+[6]: https://arxiv.org/pdf/2310.14821
+
+Staking
+-------
+
+The marketplace smart contract only slashes a storage provider when there is a
+combined BLS signature that represents > 2/3 stake.
+
+It can be expensive to calculate the amount of stake associated with a combined
+BLS signature on-chain. Because any combination that represents > 2/3 stake is
+valid, there can be many different valid combinations. If we have to calculate
+the amount of stake every time that a validator submits a combined signature
+that signals that a proof was missed, then the gas fees would be prohibitive.
+
+So instead we expect there to be pre-calculated combined public keys that
+represent > 2/3 stake majorities. The gas costs for validating the stake that
+these combined public keys represent can be borne by the validators when they
+put down their stake.
+
+Pros and cons
+-------------
+
+There are a couple of advantages to this design in which we use a consensus
+protocol to sequence transactions.
+
+A storage provider only needs to send its proofs to a single validator, and
+the consensus protocol ensures that all validators see it.
+
+There is no need for validators to sign off on individual storage proofs. The
+[Mysticeti paper][6] points out that signing and verifying signatures is one of
+the main contributors to latency in a consistent broadcast design. By only
+signing entire blocks of transactions this is avoided in the Mysticeti protocol.
+
+This design is suitable for supporting other kinds of transactions later on,
+such as payments, payment channels, and marketplace interactions.
+
+Compared to a [previous iteration][7] of this design we have one less role.
+There no longer is a separate role that monitors the on-chain marketplace to
+check which storage proofs are required. Also, there is no longer a race to go
+on-chain to mark a proof as missing. Because we have a consensus protocol we can
+select which validator goes on-chain.
+
+In exchange for these advantages we have some drawbacks as well.
+
+The number of validators is by necessity fairly small (in the order of < 100
+validators) because of the communication between the validators. Measures can be
+taken to increase the decentralization of the validators. We could for example
+introduce epochs in which some validators are chosen from a larger set of
+potential validators, but that comes at the expense of added complexity.
+
+The latency is larger than in a [previous iteration][7] of this design, because
+the consensus protocol requires 3 rounds of communication before it has
+sequenced the transactions. That might be mitigated by using the fast path from
+the Mysticeti protocol for `StorageProof` transactions.
+
+[7]: https://github.com/codex-storage/codex-research/pull/194
```

Comment:
I see many other drawbacks to this approach. 

* What motivates a node to become a validator? If everything goes well, they do a big part of the work for no reward. 
* How do we avoid SPs and validators colluding?

Reply:
> * What motivates a node to become a validator? If everything goes well, they do a big part of the work for no reward.

I intend to add fees for including transactions.

> * How do we avoid SPs and validators colluding?

Through having multiple staked validators. The scheme works as long as >2/3 stake is not byzantine.